src/components/SeatMap/api.test.js

```diff
@@ -0,0 +1,196 @@
+import { JetsSeatMapApiService } from './api';
+import {
+  cabin,
+  entertainment,
+  power,
+  wifi,
+  seatDetails,
+  cabinItem,
+} from './__fixtures__/seatMapApiPostDataResponse';
+import { flightDetails } from './__fixtures__/seatMapApiGetPlaneFeatures';
+
+jest.mock('./api', () => {
+  const module = jest.requireActual('./api');
+  return {
+    ...module,
+    JetsSeatMapApiService: class extends module.JetsSeatMapApiService {
+      postData = jest.fn();
+    },
+  };
+});
+
+const api = new JetsSeatMapApiService('appId', 'key', 'url');
+
+describe('JetsSeatMapApiService', () => {
+  describe('when the getPlaneFeatures function is called', () => {
+    beforeEach(() => {
+      const singleCabinResponseFixture = [{
+        id: '1111',
+        cabin: cabin(),
+        entertainment: entertainment(),
+        power: power(),
+        wifi: wifi(),
+        seatDetails: seatDetails(),
+      }];
+      api.postData.mockImplementation(() => singleCabinResponseFixture)
+    })
+
+    it('should call postData with the correct parameters', async () => {
+      const flightFixture = flightDetails();
+
+      await api.getPlaneFeatures(flightFixture, 'EN', 'metric');
+
+      expect(api.postData).toHaveBeenCalledWith(
+        'flight/features/plane/seatmap',
+        { flight: flightFixture, lang: 'EN', units: 'metric' },
+      );
+    });
+
+    it('should call postData with the default language if the provided language is not supported', async () => {
+      const flightFixture = flightDetails();
+
+      await api.getPlaneFeatures(flightFixture, 'A', 'metric');
+
+      expect(api.postData).toHaveBeenCalledWith(
+        'flight/features/plane/seatmap',
+        { flight: flightFixture, lang: 'EN', units: 'metric' },
+      );
+    })
+  });
+
+  describe('for a specific cabin class (F, B, P or E)', () => {
```

Comment:
Could we update this description to be 'when a specific cabin class (F, B, P or E) is provided' please? This is to try to be consistent with the when/should pattern